src/template.rs

```diff
@@ -0,0 +1,81 @@
+use handlebars::{handlebars_helper, Handlebars};
+use rand::seq::SliceRandom;
+
+pub fn addhelpers(x: &mut Handlebars) {
+    handlebars_helper!(wrap: |index: usize, s: String| {
+        let mut data = s.clone();
+        let mut wraploc: usize = 0;
+        let mut lastwrap: usize = 0;
+
+        for i in 0..(s.len() - 1) {
+            wraploc += 1;
+            if wraploc >= index {
+                for k in (0..i).rev() {
+                    if k - lastwrap <= index && data.chars().nth(k).unwrap() == ' ' {
+                        data.remove(k);
+                        data.insert(k, '\n');
+                        lastwrap = k + 1;
+                        break;
+                    }
+                }
+                wraploc = i - lastwrap;
+            }
+        }
+        data
+    });
+    handlebars_helper!(wrap_with: |index: usize, ending: String, s: String| {
+        let mut data = s.clone();
+        let mut wraploc: usize = 0;
+        let mut lastwrap: usize = 0;
+        let mut i = 0;
+
+        while i < s.len() - 1 {
+            wraploc += 1;
+            if wraploc >= index {
+                for k in (0..i).rev() {
+                    if k - lastwrap <= index && data.chars().nth(k).unwrap() == ' ' {
+                        data.remove(k);
+                        for ch in ending.chars().rev() {
+                            data.insert(k, ch);
+                        }
+                        lastwrap = k + ending.len();
+                        i += ending.len() - 1;
+                        break;
+                    }
+                }
+                wraploc = i - lastwrap;
+            }
+            i += 1;
+        }
+        data
```

Comment:
is it possible to pull this common code between `wrap` and `wrap_with` into a function and reuse? if we think about it `wrap` is essentially `wrap_with \n`